HBASE-23891: Add an option to Actions to filter out meta RS

BukrosSzabolcs opened the pull request:
make skipping or killing meta RegionServer cofigurable for new actions

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 13s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 7 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m  3s |  master passed  |
| +1 :green_heart: |  compile  |   0m 34s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 21s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 40s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 20s |  master passed  |
| +0 :ok: |  spotbugs  |   6m  2s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   0m  0s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 13s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 32s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 32s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 18s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m 31s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 43s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 14s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m  0s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 49s |  hbase-it in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  53m 37s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.6 Server=19.03.6 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1217 |
| JIRA Issue | HBASE-23891 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 4fe92cb7311e 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1217/out/precommit/personality/provided.sh |
| git revision | master / 3c3aae9a2f |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/1/testReport/ |
| Max. process+thread count | 393 (vs. ulimit of 10000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

wchevreuil commented on hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/GracefulRollingRestartRsAction.java:
```diff
@@ -35,15 +35,23 @@
+      ServerName metaServer = cluster.getServerHoldingMeta();
+      selectedServers.remove(metaServer);
+    }
+
```
Could we move this to Action class, in order to avoid repeating it on different Action implementations? Just noticed we duplicate this in RollingBatchSuspendResumeRsAction.java.

wchevreuil commented on hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/actions/RollingBatchSuspendResumeRsAction.java:
```diff
@@ -64,6 +67,11 @@ public void perform() throws Exception {
+    if(!killMetaRs){
+      ServerName metaServer = cluster.getServerHoldingMeta();
+      selectedServers.remove(metaServer);
+    }
```
See previous comment above.

wchevreuil commented on hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/factories/ServerAndDependenciesKillingMonkeyFactory.java:
```diff
@@ -83,5 +84,8 @@ private void loadProperties() {
         MonkeyConstants.DEFAULT_ROLLING_BATCH_SUSPEND_RS_RATIO + ""));
+    killMetaRs = Boolean.parseBoolean(this.properties.getProperty(
+      MonkeyConstants.KILL_META_RS,
+      MonkeyConstants.DEFAULT_KILL_META_RS + ""));
```
Similar suggestion as done for the Actions. Can we move this to a common method in the parent class to avoid code repetition?

wchevreuil commented on hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/factories/ServerKillingMonkeyFactory.java:
```diff
@@ -79,5 +80,8 @@ private void loadProperties() {
         MonkeyConstants.DEFAULT_ROLLING_BATCH_SUSPEND_RS_RATIO + ""));
+    killMetaRs = Boolean.parseBoolean(this.properties.getProperty(
+      MonkeyConstants.KILL_META_RS,
+      MonkeyConstants.DEFAULT_KILL_META_RS + ""));
```
See previous comment.

wchevreuil commented on hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/factories/StressAssignmentManagerMonkeyFactory.java:
```diff
@@ -110,5 +111,8 @@ private void loadProperties() {
         MonkeyConstants.DEFAULT_ROLLING_BATCH_SUSPEND_RS_RATIO + ""));
+    killMetaRs = Boolean.parseBoolean(this.properties.getProperty(
+      MonkeyConstants.KILL_META_RS,
+      MonkeyConstants.DEFAULT_KILL_META_RS + ""));
```
See previous comment.

BukrosSzabolcs commented:
Hi @wchevreuil 
Thanks a lot for your feedback. 
To allow a more generic approach I made a small change and made the monkeyProperties available for Actions too. This way we can make them configurable without passing custom parameters from the ChaosMonkeys through the constructor. I moved the skip logic to the Action and now it's possible to skip the meta region server in every action with a single configuration. This made one of the existing actions redundant so I cleaned it up.

Apache-HBase commented:
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 11s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 22 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m  7s |  master passed  |
| +1 :green_heart: |  compile  |   0m 30s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 20s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 11s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 15s |  master passed  |
| +0 :ok: |  spotbugs  |   5m 27s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   0m  0s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 39s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 29s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 29s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 18s |  hbase-it: The patch generated 7 new + 122 unchanged - 1 fixed = 129 total (was 123)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m 13s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 57s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 13s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m  0s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 51s |  hbase-it in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  51m 22s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.7 Server=19.03.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1217 |
| JIRA Issue | HBASE-23891 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux d4aa0ecdd01d 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1217/out/precommit/personality/provided.sh |
| git revision | master / 9bd39786df |
| Default Java | 1.8.0_181 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/2/artifact/out/diff-checkstyle-hbase-it.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/2/testReport/ |
| Max. process+thread count | 394 (vs. ulimit of 10000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/2/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

wchevreuil commented on hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestBackupRestore.java:
```diff
@@ -139,7 +139,7 @@ public void tearDown() throws IOException {
     Policy p = new PeriodicRandomActionPolicy(sleepTime,
       new RestartRandomRsExceptMetaAction(sleepTime));
-    this.monkey = new PolicyBasedChaosMonkey(util, p);
+    this.monkey = new PolicyBasedChaosMonkey(null, util, p);
```
nit: If we just overload the constructor on PolicyBasedChaosMonkey, then we don't need this extra change just to pass a null reference.

BukrosSzabolcs replied:
You are right, I'll fix this.

wchevreuil commented on hbase-it/src/test/java/org/apache/hadoop/hbase/IntegrationTestRegionReplicaPerf.java:
```diff
@@ -195,7 +195,7 @@ public void setUpMonkey() throws Exception {
       new RestartRandomRsExceptMetaAction(sleepTime),
       new MoveRandomRegionOfTableAction(tableName));
-    this.monkey = new PolicyBasedChaosMonkey(util, p);
+    this.monkey = new PolicyBasedChaosMonkey(null, util, p);
```
See previous comment.

wchevreuil commented on hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/factories/MobNoKillMonkeyFactory.java:
```diff
@@ -75,7 +75,7 @@
     Action[] actions4 = new Action[] { new DumpClusterStatusAction() };
 
-    return new PolicyBasedChaosMonkey(util,
+    return new PolicyBasedChaosMonkey(properties, util,
```
Do we really need to specify extra properties when creating policy for all these factories? Otherwise, if this is just duo to the change in PolicyBasedChaosMonkey constructor, it would benefit from overloading the constructor there.

BukrosSzabolcs replied:
I kind of had to. The issue was that Actions do not have access to these properties. This is the main reason we have this clutter in factories where we read the properties and pass them along to the actions in the action constructor, when it could have been an internal thing. So I looked for a way to get these properties to them. (It was a minor issue that Action.init could not use these properties and had to rely on the default values from the config. Passing it along fixed this too.) Anyway the best way to do this was through PolicyBasedChaosMonkey and PolicyContext. Technically I could have just created a setter on PolicyBasedChaosMonkey and leave the constructor alone and call the setter from IntegrationTestBase, but it felt like a bad practice to circumvent a factory.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  3s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 22 new or modified test files.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 57s |  master passed  |
| +1 :green_heart: |  compile  |   0m 29s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 20s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 15s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 15s |  master passed  |
| +0 :ok: |  spotbugs  |   5m 32s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   0m  0s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 34s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 19s |  hbase-it: The patch generated 0 new + 122 unchanged - 1 fixed = 122 total (was 123)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m 15s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 35s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 13s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m  0s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 52s |  hbase-it in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  50m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.7 Server=19.03.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1217 |
| JIRA Issue | HBASE-23891 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 201860f0f170 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1217/out/precommit/personality/provided.sh |
| git revision | master / 9bd39786df |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/3/testReport/ |
| Max. process+thread count | 394 (vs. ulimit of 10000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/3/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 47s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --findbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m 33s |  master passed  |
| +1 :green_heart: |  compile  |   0m 30s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 30s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 14s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m  2s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 31s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 31s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 20s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 12s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 53s |  hbase-it in the patch passed.  |
|  |   |  28m 48s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.7 Server=19.03.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/4/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1217 |
| JIRA Issue | HBASE-23891 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux d61998cc957c 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9c67abef18 |
| Default Java | 1.8.0_232 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/4/testReport/ |
| Max. process+thread count | 392 (vs. ulimit of 10000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/4/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 48s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  3s |  Unprocessed flag(s): --brief-report-file --findbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 51s |  master passed  |
| +1 :green_heart: |  compile  |   0m 40s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m 44s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 15s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   7m 20s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 36s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 36s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   5m 44s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 13s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 55s |  hbase-it in the patch passed.  |
|  |   |  32m 18s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.7 Server=19.03.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/4/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1217 |
| JIRA Issue | HBASE-23891 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 3849c7b9c56b 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9c67abef18 |
| Default Java | 2020-01-14 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/4/testReport/ |
| Max. process+thread count | 423 (vs. ulimit of 10000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/4/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

Apache-HBase commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 15s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m  5s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 19s |  master passed  |
| +0 :ok: |  spotbugs  |   0m 29s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   0m  0s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 37s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 18s |  hbase-it: The patch generated 0 new + 122 unchanged - 1 fixed = 122 total (was 123)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  hadoopcheck  |  17m 54s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  findbugs  |   0m  0s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   0m 11s |  The patch does not generate ASF License warnings.  |
|  |   |  38m 55s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.7 Server=19.03.7 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/4/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1217 |
| JIRA Issue | HBASE-23891 |
| Optional Tests | dupname asflicense spotbugs findbugs hadoopcheck hbaseanti checkstyle |
| uname | Linux df0fa84e8ffa 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | master / 9c67abef18 |
| Max. process+thread count | 63 (vs. ulimit of 10000) |
| modules | C: hbase-it U: hbase-it |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1217/4/console |
| versions | git=2.17.1 maven=2018-06-17T18:33:14Z) |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.
